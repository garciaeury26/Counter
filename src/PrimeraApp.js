import React from 'react'
import PropTypes from 'prop-types'

export const PrimeraApp = ({ saludo, subtitulo }) => {
    return (
        <div>
            <h1 style={{ color: '#fff' }}>{saludo}</h1>
            <span>{subtitulo}</span>
        </div >
    )
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy el jefe'
}
