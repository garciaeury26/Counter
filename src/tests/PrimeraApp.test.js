import React from 'react';
import '@testing-library/jest-dom'

//*si con jest normal
// import '@testing-library/jest-dom'
// import { render } from '@testing-library/react';

//* si em para enzyme
import { shallow } from 'enzyme'

import { PrimeraApp } from '../PrimeraApp';

describe('Pruabas en el componente de pruebas', () => {

    // test('debe de mostrar el mensaje soy el jefe => del subtitulo', () => {
    //     const saludo = 'hi my name is eury';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    const saludo = 'hi my name is eury';
    const subtitulo = 'nice to meet you'

    test('Deve de mostrar <PrimeraApp/> corractamente', () => {
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Deve de mostrar el subtitulo enviado por props', () => {
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        //buscate un span en el componente
        //* pudes llamar lo tambien con id
        //hay mas propiedades en este caso solo el texto de la etiqueta
        const textoParrafo = wrapper.find('span').text();

        expect(textoParrafo).toBe(subtitulo);

    })




})
