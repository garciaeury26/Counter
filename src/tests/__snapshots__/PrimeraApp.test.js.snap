// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruabas en el componente de pruebas Deve de mostrar <PrimeraApp/> corractamente 1`] = `
<div>
  <h1
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    hi my name is eury
  </h1>
  <span>
    Soy el jefe
  </span>
</div>
`;
